src/gui/book/Book.cpp

```diff
@@ -1696,5 +1672,21 @@ void MapPage::ARX_INTERFACE_ManageOpenedBook_LeftTabs_Map() {
 }
 
 void JournalPage::render() {
-	manageQuestBook();
+	// Cache the questbook data
+	if(questBook.text().empty() && !g_playerQuestLogEntries.empty()) {
+		std::string text;
+		for(size_t i = 0; i < g_playerQuestLogEntries.size(); ++i) {
+			std::string quest = getLocalised(g_playerQuestLogEntries[i]);
+			if(!quest.empty()) {
+				text += quest;
+				text += "\n\n";
+			}
+		}
+		questBook.setData(Note::QuestBook, text);
+		questBook.setPage(questBook.pageCount() - 1);
+	}
+
+	manageNoteActions(questBook);
```

Comment:
This is all not really about rendering but about managing the state of the quest book.